### ISSUE TITLE
music_pack_path is misleading; there is only one mapping

### ISSUE THREAD

**Issue by jrhacker** [chocolate-doom/chocolate-doom#993]
3.0.0: i_sdlmusic.c LoadSubstituteConfigs loads and merges a hardcoded list
of music substitution mapping files from either music_pack_path or
configdir/music: doom1-music.cfg, doom2-music.cfg, tnt-music.cfg,
heretic-music.cfg, hexen-music.cfg, strife-music.cfg.

In reality, there is only one mapping, from sha1 to music file.  When music
is reused, different mappings in different files just override one another.

It would be simpler and less confusing just to load a single music.cfg from
configdir.  Attaching sample for doom.wad, doom2.wad, tnt.wad, plutonia.wad.

[music.cfg.txt](https://github.com/chocolate-doom/chocolate-doom/files/1736662/music.cfg.txt)

**Comment by jmtd** [comment]
I'm inclined to agree. I've only recently got music packs working at all (completely failed first time I tried).

**Comment by fragglet** [comment]
The reason that it's structured this way is so that music packs can distribute a separate config file for just that particular set of replacements. For example see here where there are separate music packs for Doom, Doom II and Final Doom:

http://sc55.duke4.net/games.php#doom

**Comment by jrhacker** [comment]
When you have PWADs that bring their own music, like Icarus and the Plutonia MIDI Pack, their mapping has to be merged with one of the 6 files that are already spoken for.  To keep it modular, it would be necessary to load arbitrary config files found in music_pack_path, not just a hardcoded list.

**Comment by fragglet** [comment]
Yes, if you're proposing that a glob on `*.cfg` would be better, I agree.

**Comment by jrhacker** [comment]
When different config files compete to define the mapping for a given hash, it may not be obvious which one wins.  But *.cfg is less worse than a harcoded list.

**Comment by fragglet** [comment]
Yeah, I'm not sure what the behavior ought to be when different config files compete for the same hash. Some thoughts though:

* It should probably be an error for the same file to list multiple entries for the same hash (at least).
* Exiting with an error if two files conflict with each other seems excessively user hostile. It would be better to print a warning and resolve automatically
* Resolving in a deterministic manner seems better (eg. use file modification time, or filename), but also probably more work than it's worth
* As long as there's a warning message it's probably not a huge deal
